Patch 1 (#246)
CleverCloud/documentation: content/doc/addons/keycloak.md

---
type: docs
title: Keycloak

> shortdesc: This add-on provides a one-click Keycloak single sign-on and identity and access management (IAM) solution based on existing Clever Cloud services and taking advantage of the *Please open-it* experiences, our Keycloak expert partner.

Reviewer: `shortdesc`-> `description`

Reviewer 2: ```suggestion
description: This add-on provides a one-click Keycloak single sign-on and identity and access management (IAM) solution based on existing Clever Cloud services and taking advantage of the *Please open-it* experiences, our Keycloak expert partner.
```

![Open the application button]({{< ref `assets/images/open_the_app_ico.png` >}})

You also can get the Keycloak URL from the Clever Cloud console, in the information panel of your Keyckoak add-on.

### Keycloak admin interface
The admin interface is accessible on the `/admin` endpoint:

- `iffslonattslbta6eilb-keycloak.services.clever-cloud.com/admin`
> Here, the URL comes from the information panel in the console and we add `/admin` at the end.

### Admin User and Password
- A default user and password is provided in the Java application's environment variables
- __At your first connection, you will have to change the password__

### Realms management
We recommend using __as few Realm as possible__ : they may significantly impact the overall performances.

Realms can be managed in two way
1. __RECOMMENDED:__ using Clever Cloud's Keycloak add-on environment variables
- In your Keycloak java application, go to the `Environment variables` panel and set as many Realms you need, separate by `,`.
- Rebuild and redeploy your Java application after the change

> > to deploy two realm `realm_1`, `realm_2` just set `REALMS=realm_1,realm_2`

Reviewer 2: ```suggestion
> to deploy two realms `realm_1`, `realm_2` just set `REALMS=realm_1,realm_2`
```

- A cold export can be done at the start

> This is a total export, including the client's secrets and hashed password

- Set in the environment variables tab of the Keycloak Java application from the Clever Cloud console:
    - `CC_EXPORT_REALM`
- restart the Java application from the Clever Cloud console

### Importing data
Importing Realm is done by adding priory exported in the `Import` folder in the associated FSBucket. After restarting the Java application, they would be imported.

## Security and updates
The Keycloak add-on is a fully managed application, you don't have to select a particular version. Still, it receives updates for both features and security, that we will manage for you with continuously upgraded versions over time.
After being updated, your Keycloak add-on could need to be restarted.

## Pricing plans
### Technical Review price plan
Keycloak on Clever Cloud is the easiest way to set up a ready-to-use Keycloak.

It starts with
- Java 17: XS
- PostgreSQL: XS Tiny Space
- FSBucket

We add services management fees for maintenance, configuration, platform costs, and support. During the technical review, this service management is free.

> Hence, the total price by month for the base plan is (according to the [pricing page](https://www.clever-cloud.com/pricing/)):

Reviewer 2: ```suggestion
Hence, the total price for 30 days for the base plan is (according to the [pricing page](https://www.clever-cloud.com/pricing/)):
```

> - `JAVA XS` (16€) + `PostgreSQL XS Tiny Space` (15€) + `< 100MB FSBucket` (0€) + `add-on management fees` (O€) = 31€/month

Reviewer 2: ```suggestion
- `JAVA XS` (16€) + `PostgreSQL XS Tiny Space` (15€) + `< 100MB FSBucket` (0€) + `add-on management fees` (O€) = 31€/30 days
```
